### ISSUE TITLE
Scroll bars for legend widgets won't move coherently with displayed data 

### ISSUE THREAD

**Issue by ale-cristofori** [geosolutions-it/MapStore2#10070]
## Description
The scroll bar appearing on the right part of the legend widget, and potentially in all widgets, if moved with the mouse wheel will jump to the following results, skipping the display of some of the legend items. The scrolling is not 'smooth' enough to allow the display of all legend items and some of them are skipped from display.

## How to reproduce
This issue is reproducible when there is a piece of data with a lot of unique values to be included in the legend of the widget, for example the pie chart.
- Select a data that has the above characteristics and create a pie chart widget for it
- Include a legend
- Notice how, when the window that include the chart is under a certain size, the scroll bar that scrolls through the legend will 'jump' (using the mouse scroll wheel) without scrolling evenly. This hides some legend items in the legend through scrolling.

https://github.com/geosolutions-it/MapStore2/assets/14953970/6c580bd9-e1fd-4559-b161-5c00087de1a3

*Expected Result*
The legend, scrolling with the mouse wheel, should scroll through evenly without jumping

*Current Result*
The legend, scrolling with the mouse wheel, the scroll bar that scrolls through the legend will 'jump' (using the mouse scroll wheel) without scrolling evenly. This hides some legend items in the legend through scrolling.

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

## Other useful information
The behaviour is something that has to do with plotly. As signalled in the below issues
https://github.com/plotly/plotly.js/issues/6737
https://community.plotly.com/t/scrollbar-sensitivity-in-plot-legend-skips-curves/78866

**Comment by dsuren1** [comment]
@ale-cristofori Thanks for this export of the test data [map_specie_alberi.json](https://github.com/geosolutions-it/MapStore2/files/15274524/map_specie_alberi.json) but unfortunately, the services required by these layers are not accessible by me as I couldn't connect to FIRENZE vpn. 

@tdipisa Is it possible add this layer to the existing [csw](https://gn3-rndt.geo-solutions.it/geonetwork/srv/eng/csw)? or maybe point me to a map or a layer which can produce many legends on a chart ?

**Comment by tdipisa** [comment]
> @ale-cristofori Thanks for this export of the test data but unfortunately, the services required by these layers are not accessible by me
> 
> @tdipisa Is it possible to add this layer to the existing [csw](https://gn3-rndt.geo-solutions.it/geonetwork/srv/eng/csw)?
> 
> or @ale-cristofori @tdipisa maybe point me to a map or a layer in MS which can produce many legends on a chart ?

@dsuren1 use this map which contains the sample dataset you need.

**Comment by dsuren1** [comment]
### Investigation outcome
As reported in the description, this is an [open issue](https://github.com/plotly/plotly.js/issues/6737) in plotly.js and we do not have a set timeline for its resolution yet.

However, we can handle this case in MS on chart initialization by adding an event listener of the wheel event in capture mode and dispatch a modified event with new `deltaY` value, such that the sensitivity of the wheel event is exhibited in a controlled manner

_Demo with a rudimentary fix implemented_

https://github.com/geosolutions-it/MapStore2/assets/26929983/c1786044-7124-45a0-a955-db42e6b967ff

### Suggestion 
Instead of adding a predefined sensitivity value, we can also expose this as a configurable property for the chart. ex: `legendMouseWheelSensitivity`. Your thoughts @tdipisa ?

**Comment by tdipisa** [comment]
Thank you @dsuren1. I would like you to share the above with other core team members (like @allyoucanmap or @offtherailz) since I think it is something probably quick but that we should handle carefully if we have to implement a kind of workaround in MS.

**Comment by dsuren1** [comment]
@allyoucanmap 
As per our discussion,
I tested with multiple browsers (Chrome, Firefox, Opera), and found that normalizing the `deltaY` property results in consistent scrolling behavior across all of them. It appears that a fixed value of `deltaY: ±0.30` can effectively normalize scrolling behavior, with positive and negative deltaY values corresponding to upward and downward scrolling, respectively.

> [!WARNING]
During this I observed an existing behavior with Safari where the mouse wheel scroll doesn't work on legend in Chart.
